content/blog/2016/2016-08-03-docker-pipeline-environments.adoc

```diff
@@ -0,0 +1,120 @@
+---
+layout: post
+title: "Define your environment with Docker Pipeline, don't install software"
+tags:
+- jenkins
+- dsl
+- pipeline
+- plugins
+- blueocean
+- ux
+- javascript
+- nodejs
+author: michaelneale
+---
+
+If you are running parts of your pipeline on Linux, possibly the easiest way to get 
+a clean reusable environment is to use: link:https://go.cloudbees.com/docs/cloudbees-documentation/cje-user-guide/chapter-docker-workflow.html[Docker Pipeline].
+
+In this short post I wanted to show how you can avoid installing stuff on the agents, and have per project, or even per branch, customized build environments. 
+Your environment, as well as your pipeline is defined and versioned alongside your code. 
+
+This is used in the "dogfood" project for Blue Ocean. (see link:https://en.wikipedia.org/wiki/Eating_your_own_dog_food[eating own dogfood]). Some call it "drinking your own Champagne" but unless you live in France technically you can't say that.
+
+
+=== Environment and Pipeline for Javascript components
+
+The link:/doc/blueocean[Blue Ocean] project has a few moving parts, one of which is called the "Jenkins Design Language". 
+This is a grab bag of re-usable css, html, style rules, icons and javascript components (using React.js) that provide the look and feel that 
+makes Blue Ocean look the way it does (it's intention isn't to be Blue Ocean specific).
+
+Javascript and the web being what it is in 2016, it is in need of various tooling. 
+This includes npm and all that it needs. Less to convert less to css, Babel to "transpile" versions of Javascript to other types of Javascript (don't ask) and more. 
+
+We could spend time installling nodejs/npm on the agents, but why not just use the link:https://hub.docker.com/_/node/[official off the shelf] docker image from docker hub? 
+
+This way the only thing that has to run on the build agents, is the Jenkins agent, and a docker daemon. This gave a good start of a pipeline in the `Jenkinsfile` in the root of the repo:
```

Comment:
"This way the only thing that has to run on the build agents"? 
You mean something like "The only thing that has to be installed and run on the build agents"?  

"gave a good start of"?